tests/conftest.py

```diff
@@ -281,6 +267,30 @@ def _fc(request, fc_module, new_user=False):
     return fc_module
 
 
+def _port(port, busy_ports, ip=_LOCALHOST):
+    import socket
+    from pykern.pkdebug import pkdlog
+
+    p = int(port)
+    while sirepo.const.PORT_MIN <= p <= sirepo.const.PORT_MAX:
```

Comment:
This should be in in a smaller range that's outside the normal range for manually started servers, e.g. 10000-11000.  This can be called "TEST_START". If we are testing 30K+ ports, there's a bug. Narrowing the range avoids too long of a wait.

Start at a random point within the range so that we don't need to keep "busy_ports" state.

Reply:
[739348c](https://github.com/radiasoft/sirepo/pull/4846/commits/739348cec891a109dbd3ad4390a8c9390d92ba7a)